pkg/cmd/factory/default_test.go

```diff
@@ -362,9 +363,25 @@ func Test_ioStreams_prompt(t *testing.T) {
 			config:         disablePromptConfig(),
 			promptDisabled: true,
 		},
+		{
+			name:           "prompt disabled via GH_PROMPT_DISABLED env var",
+			env:            map[string]string{"GH_PROMPT_DISABLED": "1"},
+			promptDisabled: true,
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			if tt.env != nil {
+				for k, v := range tt.env {
+					old := os.Getenv(k)
+					os.Setenv(k, v)
+					if k == "GH_PROMPT_DISABLED" {
+						defer os.Unsetenv(k)
+					} else {
+						defer os.Setenv(k, old)
+					}
```

Comment:
This can now all be simplified by using `t.Setenv` a new feature in Go 1.17.

```suggestion
t.Setenv(k,v)
```

Reply:
nice! I didn't know about that one!